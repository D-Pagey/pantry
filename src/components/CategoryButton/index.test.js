import React from 'react';
import userEvent from '@testing-library/user-event';
import { CategoryButton } from '.';

const props = {
    name: 'meat'
};

describe('CategoryButton component', () => {
    it('should render', () => {
        const { container } = render(<CategoryButton {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onClick when clicked', () => {
        const handleClick = jest.fn();

        const { getByText } = render(<CategoryButton {...props} handleClick={handleClick} />);

        userEvent.click(getByText('Meat'));

        expect(handleClick).toHaveBeenCalled();
    });
});
