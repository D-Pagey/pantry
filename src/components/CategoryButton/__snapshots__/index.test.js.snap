// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border: 1px solid rgba(52,136,65,0.6);
  border-radius: 16px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 115px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 144px;
}

.c1 {
  color: #348841;
  font-size: 18px;
}

<button
  class="c0"
  data-testid="categoryCardmeat"
  type="button"
>
  <p
    class="c1"
  >
    Meat
  </p>
</button>
`;
